src/address/locale-address.ts

```diff
@@ -0,0 +1,17 @@
+import { Country } from '../geography';
+
+import Address from './address';
+
+export default function localeAddress<T1 extends Address>(address: T1, countries?: Country[]): T1 {
+    const country =  (countries || []).find(({ code }) => code === address.countryCode);
+    const states = !country || !country.subdivisions.length ? [] : country.subdivisions;
+    const state = states.find(({ code }) => code === address.stateOrProvinceCode);
+
+    return Object.assign({},
+        address,
+        {
+            country: country ? country.name : address.country,
```

Comment:
since this could be a breaking change, maybe expose this new data via new fields? `localedCountry` and `localedStateOrProvince` ?